src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
I have a use case for that. I want to load my full configuration into a domain object. Within the full configuration, there's an object with Spark properties. My application keys are hyphen separated in the configuration file, but the Spark keys retain their original name, which uses [camel case](http://spark.apache.org/docs/latest/configuration.html#application-properties). I want to retain the original names of the Spark properties because I want to build a `Config` from that domain object later on, whose values are to be dealt directly by Spark.

We can simplify the interface by just requiring a function `String => String` and I can still implement my use case with a custom `ConfigConvert` for my Spark properties class, so that's not a big issue. In that case, do you think it still makes sense for `pureconfig` to provide those helpers for different word delimiters, or should we leave that completely for the user to deal with?